verticals/vertical-crm/providers/salesforce/updatePermissions.ts

```diff
@@ -0,0 +1,103 @@
+import * as jsforce from 'jsforce'
+
+interface SalesforceInstance {
+  getJsForce: () => Promise<jsforce.Connection>
+}
+
+const API_VERSION = '59.0'
+
+export async function updateFieldPermissions(
+  instance: SalesforceInstance,
+  objectName: string,
+  nonPrimaryFields: string[],
+) {
+  // After custom fields are created, they're not automatically visible. We need to
+  // set the field-level security to Visible for profiles.
+  // Instead of updating all profiles, we'll just update it for the profile for the user
+  // in this connection.
+  //
+  // We're doing this all the time, even if there were no detected added fields, since
+  // the previous call to this endpoint could have failed after creating fields but before
+  // adding permissions, and we want the second call to this endpoint to fix that.
+  //
+  // TODO: do we want to make it visible for all profiles?
+  const sfdc = await instance.getJsForce()
```

Comment:
this function would be cleaner if we pass the `jsforce.Connection` instance because it can be re-used rather than needing to be refetched each time which is expensive.